feat: adding forbid-mutex setting (#12)
manuelarte/embeddedstructfieldcheck: README.md

  enable:
    - embeddedstructfieldcheck 
    ...

  settings:
    embeddedstructfieldcheck:

>       # Checks that sync.Mutex is not used as embedded field.

Reviewer: Maybe

```suggestion
      # Checks that mutex (`sync.Mutex` and others) is not used as embedded field.
```

And then use it as:

```bash
embeddedstructfieldcheck [-forbid-mutex] [--fix]
```

- `forbid-mutex`: `true|false` (default `false`)
   Checks that `sync.Mutex` and `sync.RWMutex` are not used as embedded fields.
- `fix`: `true|false` (default `false`)
   Fix the case when there is no space between the embedded fields and the regular fields.

> ## Why not using `sync.Mutex` as embedded field
> 
> You should not expose your internal synchronization to the callers.
> By embedding `sync.Mutex` you're exposing `Lock` and `Unlock` methods when you should not.

Reviewer: In the paragraph above, you are mentioning sync.Mutex and sync.RWMutex

So I would suggest this

```suggestion
## Why not using `sync` mutex as embedded field

You are granting access to your internal synchronization methods out of your struct.

This should not be delegated out to the callers. It's a source of bugs.
```

And then use it as:

```bash

> embeddedstructfieldcheck [-forbid-mutex=true|false] [--fix]
> ```
> 
> - `forbid-mutex`: `true|false` (default `false`) Checks that `sync.Mutex` is not used as an embedded field.
> - `fix`: Fix the case when there is no space between the embedded fields and the regular fields.

Reviewer: 🎨 minor stylistic idea

`fix` also supports using `-fix=true/false`

What about this?

````suggestion
embeddedstructfieldcheck [-forbid-mutex] [--fix]
```

- `forbid-mutex`: `true|false` (default `false`) Checks that `sync.Mutex` is not used as an embedded field.
- `fix`: `true|false` (default `false`) Fix the case when there is no space between the embedded fields and the regular fields.
````

And then maybe thus would be clearer

````suggestion
embeddedstructfieldcheck [-forbid-mutex] [--fix]
```

- `forbid-mutex`: `true|false` (default `false`)
   Checks that `sync.Mutex` is not used as an embedded field.
- `fix`: `true|false` (default `false`)
   Fix the case when there is no space between the embedded fields and the regular fields.
````

> ## Resources

Reviewer: You might add a link in resources

> Exception: Mutexes should not be embedded, even on unexported types. 

https://github.com/uber-go/guide/blob/master/style.md#embedding-in-structs

I feel like forbid-mutex argument is only documenting what is does, but not the why it might be interesting to use it
